main/inc/lib/extra_field.lib.php

```diff
@@ -3110,6 +3124,34 @@ public function searchOptionsFromTags($from, $search, $options)
         return $result;
     }
 
+    /**
+     *  Gets the display name of an extra field.
+     *
+     * @param string $variableName
+     */
+    public static function getDisplayNameByVariable($variableName = null)
+    {
+        if ($variableName == null) {
+            return null;
+        }
+        $variableName = Security::remove_XSS($variableName);
```

Comment:
remove_XSS shoul be at the time of display, not before inserting into the database (because it can have side effects and because the data might not always be printed in HTML - for example if you want a CSV export and remove_XSS has somehow modified the input, this could affect your results)